core-common/src/main/java/org/apache/kylin/common/KylinConfigBase.java

```diff
@@ -206,12 +207,15 @@ protected Properties getProperties(Collection<String> propertyKeys) {
     }
 
     protected StrSubstitutor getSubstitutor() {
-        // env > properties
-        final Map<String, Object> all = Maps.newHashMap();
-        all.putAll((Map) properties);
-        all.putAll(System.getenv());
+        return new StrSubstitutor(getPropertiesMap());
+    }
 
-        return new StrSubstitutor(all);
+    protected Map<String, Object> getPropertiesMap() {
+        if (propertiesMap.isEmpty()) {
+            propertiesMap.putAll((Map) properties);
+            propertiesMap.putAll(System.getenv());
+        }
```

Comment:
When calling 'getPropertiesMap()' at the first time, it put all values of properties and System.getenv() into propertiesMap, but if there are some values be changed, the values in propertiesMap cann't be changed too?